Task 7: Vote on an article

sausagroll90 opened the pull request.

pippaamy commented on __tests__/app.test.js:
```diff
@@ -86,45 +86,148 @@ describe("/api/articles", () => {
+        created_at: "2020-11-03T09:12:00.000Z",
+        votes: -5,
+        article_img_url:
+          "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
```
very thorough testing, great work!

pippaamy commented on endpoints.json:
```diff
@@ -25,6 +25,25 @@
+    },
+    "exampleResponse":{
+      "article": {
+        "article_id": 7,
```
nice!

pippaamy commented on model/articles.model.js:
```diff
@@ -27,3 +27,17 @@ exports.selectArticleById = async (id) => {
+    `UPDATE articles
+    SET votes=votes+$1
+    WHERE article_id=$2
+    RETURNING article_id, title, topic, author, body, created_at, votes, article_img_url`,
```
looks great, well done!

pippaamy reviewed (commented):
Great work, very well done!!

sausagroll90 committed:
Feature: Add article voting

Can now vote on an article by pressing the "Vote Up" and "Vote Down" buttons.
Each button is disabled when pressed, and undoes the other button if the other button has been pressed.
Potential issue, can't undo a vote without voting in the other direction.

pippaamy reviewed (commented):
The vote is working great whilst the user is online. When the user is offline it looks like their vote has gone through but on refresh it hasnt. Can we change it so if there is an error/users offline that the number changes back without refreshing :)
